src/backend/catalog/index.c

```diff
@@ -3121,8 +3121,15 @@ index_build(Relation heapRelation,
 	 * only btree has support for parallel builds.
 	 *
 	 * Note that planner considers parallel safety for us.
+	 *
+	 * SINGLENODE_FIXME: Disable parallel index building for now.
+	 * Parallel is not supported for singlenode currently, but it can still use
+	 * parallel index while create table.
+	 *
+	 * In the future, it should be considered to introduce parallelism to singlenode
+	 * mode, even including utility mode.
 	 */
-	if (parallel && IsNormalProcessingMode() &&
+	if (parallel && GP_ROLE_SINGLENODE != Gp_role && IsNormalProcessingMode() &&
```

Comment:
should also disable utility mode too. If login in with utility mode, and build index with parallel will cause crash because of accessing null pointer.

Reply:
Good point. I think we aren't ready to use parallel in utility mode yet. Updated.

Reply:
I think we do not need utility mode on singlenode, we can forbid utility mode.